graphql_compiler/compiler/filters.py

```diff
@@ -641,4 +642,14 @@ def process_filter_directive(filter_operation_info, context):
         raise GraphQLCompilationError(u'The filter with op_name "{}" must be applied on a field. '
                                       u'It may not be applied on a type coercion.'.format(op_name))
 
+    context['metadata'].record_filter_info(
+        location,
+        FilterInfo(
+            fields=(filter_operation_info.field_name,) if op_name != 'name_or_alias' else
```

Comment:
Extract the `fields` value into a variable? This one statement is getting pretty complicated now.

Bonus points if you make some constants for the filter names.

Reply:
Extracted it. I would prefer to leave filter name constants to another patch because if I do that, I can change the rest of the code to use them too. That could simplify/cleanup some things which is nice. But it is conceptually somewhat separate from what this patch is doing. So I don't necessarily want to mix both.